plugins/destination/clickhouse/client/client.go

```diff
@@ -20,30 +19,26 @@ type Client struct {
 	spec     *Spec
 
 	logger zerolog.Logger
-	mode   specs.MigrateMode
-	destination.UnimplementedUnmanagedWriter
+	writer *batchwriter.BatchWriter
+	plugin.UnimplementedSource
 }
 
-var _ destination.Client = (*Client)(nil)
+var _ plugin.Client = (*Client)(nil)
+var _ batchwriter.Client = (*Client)(nil)
 
-func (*Client) DeleteStale(context.Context, schema.Tables, string, time.Time) error {
-	return errors.New("DeleteStale is not implemented")
+func (*Client) DeleteStale(context.Context, []*message.WriteDeleteStale) error {
+	return plugin.ErrNotImplemented
 }
 
 func (c *Client) Close(context.Context) error {
 	return c.conn.Close()
```

Comment:
```suggestion
	if err := c.writer.Close(ctx); err != nil {
		return err
	}
	return c.conn.Close()
```

Reply:
but I think it should always issue conn close, too.
If we were on go 1.20 I'd use `errors.Join`, but I'd make a workaround for now

Reply:
64e09937466a2cce483f05bc8e5da59badd933d0